gokce_fingerprint_verification.py

```diff
@@ -0,0 +1,184 @@
+######################################################################
+# Names of any others you worked with:
+# AI transcript if used: Claude at moments I got really stuck.
+# Any extensions done:
+# Name: Gokce Can Ertem
+######################################################################
+
+# Function to read the file and turn it into a fingerprint
+def read_and_list_fprint(filename):
+    """Reads the given print and returns a dictionary with all info."""
+    # Your turn!
+    
+
+
+    with open (filename) as fprint:
+        #lines = fprint.read().strip().splitlines()
+        lines = fprint.read().splitlines()
+        fprint2dlist = [list(line) for line in lines]
+        #print("2d List Creating")
+        return(fprint2dlist)
+        #print(fprint2dlist)
+
+
+# Function to find the top of the fingerprint
+def find_row_with_pattern(list_2d, value, count):
+    """
+    Find the first row in a 2D list that contains at least 'count' occurrences of 'value'.
+    
+    Args:
+    list_2d (list): The 2D list to search.
+    value: The value to search for.
+    count (int): The minimum number of occurrences required.
+    
+    Returns:
+    int: The index of the first row that matches the criteria, or -1 if not found.
+    """
+    for i in range(len(list_2d)):
+        if list_2d[i].count(value) >= count:
+            return i
+
+# Function to check original and variant fingerprints
+def simple_and_variant_check(filename1, filename2, pattern_value, pattern_count, rows_to_compare=None):
```

Comment:
When something specifies that you need to define particular functions with particular inputs, you really should stick to that specification. Often, a programmer is just working on a piece of a larger project, and if you go rogue like this defining your own names and parameters, your code isn't going to mesh with anyone else's.